sentry/src/main/java/io/sentry/Scope.java

```diff
@@ -259,6 +264,41 @@ public void setUser(final @Nullable User user) {
     }
   }
 
+  /**
+   * Returns the Scope's current screen, previously set by {@link Scope#setScreen(String)}
+   *
+   * @return the name of the screen
+   */
+  @ApiStatus.Internal
+  public @Nullable String getScreen() {
+    return screen;
+  }
+
+  /**
+   * Sets the Scope's current screen
+   *
+   * @param screen the name of the screen
+   */
+  @ApiStatus.Internal
+  public void setScreen(final @Nullable String screen) {
+    this.screen = screen;
+
+    final @NotNull Contexts contexts = getContexts();
```

Comment:
A good point, but I still think it should live on the scope because it represents a current state, which should be readable by other components.
E.g. when a span gets created (e.g. by some user interaction, a slow/frozen frame, ...) we'd need a way to get the current screen for adding it as span context.

Reply:
Yeah, I'm fine with `screen` living on the scope, just meant the actual setting it to Contexts should be happening in the event processor maybe

Reply:
Alright I looked into this now, but as far as I can see EventProcessors have no direct access to the scope / a hub. But [SentryClient.applyScope](https://github.com/getsentry/sentry-java/blob/a3c77bc887d1ba2e290ffd4da391666449c7b030/sentry/src/main/java/io/sentry/SentryClient.java#L802) could be a good place for this, wdyt?

Reply:
oh you're right. But we're anyway already doing that there
```
      final Contexts contexts = sentryBaseEvent.getContexts();
      for (Map.Entry<String, Object> entry : new Contexts(scope.getContexts()).entrySet()) {
        if (!contexts.containsKey(entry.getKey())) {
          contexts.put(entry.getKey(), entry.getValue());
        }
      }
```
so I think it's fine to keep it as you initially implemented 👍